include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -0,0 +1,330 @@
+/**
+ * @file behavior_manager.h
+ * @author Ethan J. Musser (emusser@seas.upenn.edu)
+ * @brief Container for managing the `Behaviors` available to a robot and
+ *        switching between them
+ * @date 7/26/22
+ * 
+ * @copyright Copyright 2022 The Trustees of the University of Pennsylvania. All
+ *            rights reserved.
+ * 
+ */
+
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <unordered_map>
+#include <utility>
+#include "kodlab_mjbots_sdk/robot_base.h"
+#include "kodlab_mjbots_sdk/behavior.h"
+#include "kodlab_mjbots_sdk/off_behavior.h"
+#include "kodlab_mjbots_sdk/log.h"
+
+namespace kodlab {
+
+/**
+ * @brief Manager for `Behavior` objects
+ * @details The behavior manager class is a container for storing numerous
+ *          behaviors for a given robot interface. It manages their selection
+ *          and transitions between them.
+ * @note By default, this class initializes its internal behaviors list with a
+ *       `kodlab::Offbehavior`. This is done for safety purposes so that, when
+ *       turned on, the robot does not command some undefined torques. This
+ *       default can be overwritten with the `SetDefaultBehavior` member.
+ * @tparam Robot[optional] derived `kodlab::RobotBase` class
+ */
+template<class Robot = kodlab::RobotBase>
+class BehaviorManager {
+  static_assert(std::is_base_of<kodlab::RobotBase, Robot>::value,
+                "Robot must have base kodlab::RobotBase.");
+
+ private:
+  /**
+   * @brief Robot that behaviors run on
+   */
+  std::shared_ptr<Robot> robot_;
+
+  /**
+   * @brief Behavior command that kills robot, default -1
+   */
+  const int KILL_ROBOT_IDX = -1;
+
+  /**
+   * @brief Name of the kill behavior
+   */
+  const std::string KILL_BEHAVIOR_NAME = "KILL";
+
+  /**
+   * @brief Behavior command that turns robot off, always 0
+   */
+  const int DEFAULT_IDX = 0;
+
+  /**
+   * @brief Name of the default behavior
+   */
+  std::string default_name_;
+
+  /**
+   * @brief Behaviors available to the robot
+   * @note The first element of this vector is always `OFF_BEHAVIOR`.  Added
+   *       behaviors are appended to the end.
+   */
+  std::vector<std::unique_ptr<Behavior<Robot>>> behaviors_;
+
+  /**
+   * @brief Map from behavior name to index in `behaviors_`
+   */
+  std::unordered_map<std::string, int> name_to_idx_map_;
+
+  /**
+   * @brief Names of behaviors in `behaviors_`
+   */
+  std::vector<std::string> names_;
```

Comment:
I'm torn. My gut says names_ that we shouldn't keep two vectors that match index the way names_ and behaviors_ do. 
Instead just query behaviors_.get_name().

It is nice for convenience, but a possible vector for errors and bugs. 
Most of the places that use names switch effortlessly. 
The member function get_behavior_names() will need a for loop or transform or something. 

The map is a useful thing to go the other way though so happy have that even though it also has a similar issue.

Reply:
This was added to avoid repeated calls to `Behavior::get_name`, particularly the looping in `get_behavior_named`.  I could go either way.